src/axom/slic/streams/LumberjackStream.hpp

```diff
@@ -230,7 +230,37 @@ class LumberjackStream : public LogStream
    *  It does not flush any messages and not all messages are guaranteed to be
    *  at the output node.
    */
-  virtual void write(bool local = false);
+  void write(bool local = false);
+
+  /*!
+   * \brief Tests whether there are any pending messages that need to be flushed.
+   * This method should only be overriden for LogStream inherited classes that can
+   * reliably test whether pending messages exist.
+   *
+   * \return Returns true if there are pending messages that need to be flushed
+   */
+  virtual bool canTestHasPendingMessages() override;
```

Comment:
Is this about whether we can test if there are pending messages, or whether it has pending messages?

Would  `canHavePendingMessages()` work for the function's name?